lib/rule-context.js

```diff
@@ -72,42 +72,27 @@ var PASSTHROUGHS = [
  * @param {object} ecmaFeatures The ecmaFeatures settings passed from the config file.
  */
 function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures) {
+    // public.
+    this.id = ruleId;
+    this.options = options;
+    this.settings = settings;
+    this.ecmaFeatures = ecmaFeatures;
+
+    // private.
+    this.eslint = eslint;
+    this.severity = severity;
```

Comment:
I tried:

|                             | Time   |
|-----------------------------|-------:|
| Assignments                 | 3010ms |
| Assignments + Object.freeze | 3070ms |
| Object.defineProperties     | 3362ms |
| Object.defineProperty x 6   | 3254ms |

I made `RuleTester` freezing those, but is it not enough?
If so, I'm going to use `Object.freeze`.